File: src/Sum Kind of Problem/C/sumkindofproblem.c
#include <stdio.h>

// 1+2+...+n = n(n+1)/2 = (n^2+n)/2
// 1+3+...+2n-1 = ((2-1) + (4-1) + ... + (2n-1)) = (2+4+...2n)-n = 2(1+2+...+n)-n = 2n(n+1)/2-n = n^2
// 2+4+...+2n = 2(1+2+...+n) = 2n(n+1)/2 = n(n+1) = n^2+n

int main()
{
	int n, i;
	scanf("%d",&n);
	while (n-- > 0)
	{
		int K, N;
		scanf("%d %d",&K,&N);
		int sq = N*N;
		printf("%d %d %d %d\n",K,(sq+N)>>1,sq,sq+N);
	}
	return 0;
}

File: Sum Kind of Problem/Go/sumkindofproblem.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scanln(&n)

	for i:=0;i<n;i++ {
		var K, N int
		fmt.Scanf("%d %d\n", &K, &N)
		sq := N * N
		fmt.Printf("%d %d %d %d\n", K, (sq+N)>>1, sq, sq+N)
	}
}

// 1+2+...+n = n(n+1)/2 = (n^2+n)/2
// 1+3+...+2n-1 = ((2-1) + (4-1) + ... + (2n-1)) = (2+4+...2n)-n = 2(1+2+...+n)-n = 2n(n+1)/2-n = n^2
// 2+4+...+2n = 2(1+2+...+n) = 2n(n+1)/2 = n(n+1) = n^2+n